### PR TITLE
[KARAF-2511] Add karaf-soap-example

### PR DESCRIPTION
**Pull request by jbonofre**

### CODE REVIEW

**Comment by jbonofre** [comment]
I added the client part. This PR is ready for a review. Thanks !

File: examples/karaf-soap-example/README.md [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+It implements a `BookingService` with a SOAP WS implementation. 
+
+The "client" bundle uses the `BookingService` with a SOAP client stub.
+
+## Artifacts
+
+* **karaf-soap-example-api** is a common bundle containing the `Booking` POJO and the `BookingService` interface.   
+* **karaf-soap-example-provider** is a blueprint bundle providing the `BookingServiceSoap` implementation of the `BookingService` interface.
+* **karaf-soap-example-client** is a CXF client to the `BookingService` SOAP.
+* **karaf-rest-example-features** provides a Karaf features repository used for the deployment.
```
Comment (fpapon) [inline examples/karaf-soap-example/README.md]: Change from karaf-rest-example-features to karaf-soap-example-feature

**Review by fpapon — COMMENTED** [review]
Fix the readme.md please

File: examples/karaf-soap-example/README.md [hunk 2]
```diff
@@ -0,0 +1,65 @@
+# Apache Karaf REST Example
```
Comment (fpapon) [inline examples/karaf-soap-example/README.md]: Change from REST to SOAP

File: examples/karaf-soap-example/README.md [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+Then, you can install the service provider feature:
+
+```
+karaf@root()> feature:install karaf-soap-example-provider
+```
+
+And the service client feature:
+
+```
+karaf@root()> feature:install karaf-soap-example-client-http
```
Comment (fpapon) [inline examples/karaf-soap-example/README.md]: Change to feature:install karaf-soap-example-client

File: examples/karaf-soap-example/README.md [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+```
+
+## Usage
+
+You can take a look on the WSDL generated for our SOAP WS:
+
+```
+http://localhost:8181/services/?wsdl
```
Comment (fpapon) [inline examples/karaf-soap-example/README.md]: Change to http://localhost:8181/cxf/services/?wsdl

**Comment by jbonofre** [comment]
Updated, ready for a new review round.

**Comment by fpapon** [comment]
LGTM

Can you squash please ?

**Comment by jbonofre** [comment]
Done, thanks.

**Pull request merged**